Challenge3/src/agent.ts

```diff
@@ -0,0 +1,55 @@
+import { Finding, HandleBlock, BlockEvent, getEthersProvider, ethers } from "forta-agent";
+import { L1_DAI_TOKEN_ADDRESS, L1_ESCROW_ABI, L1_ARB_ESCROW_ADDRESS, L1_OPT_ESCROW_ADDRESS } from "./constants";
+import { Contract } from "@ethersproject/contracts";
+import { Provider } from "@ethersproject/providers";
+import { l1Finding, getL1DaiLocked, getL2DaiSupply, emitL1Alert, l2Finding } from "./utils";
+
+let chainId: number;
+let l1OptEscrowBalance: ethers.BigNumber = ethers.BigNumber.from(0);
+let l1ArbEscrowBalance: ethers.BigNumber = ethers.BigNumber.from(0);
+let newL1OptEscrowBalance: ethers.BigNumber;
+let newL1ArbEscrowBalance: ethers.BigNumber;
+let l1DaiLocked: ethers.BigNumber;
+let l2DaiSupply: ethers.BigNumber;
+
+export function provideInitialize(provider: ethers.providers.Provider) {
+  return async function initialize() {
+    const network = await provider.getNetwork();
+    chainId = network.chainId;
+  };
+}
+
+export function provideHandleBlock(provider: ethers.providers.Provider): HandleBlock {
+  return async (blEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+
+    if (chainId === 1) {
+      const l1DaiContract = new Contract(L1_DAI_TOKEN_ADDRESS, L1_ESCROW_ABI, provider);
+      newL1OptEscrowBalance = await l1DaiContract.balanceOf(L1_OPT_ESCROW_ADDRESS, {
+        blockTag: blEvent.blockNumber,
+      });
+      newL1ArbEscrowBalance = await l1DaiContract.balanceOf(L1_ARB_ESCROW_ADDRESS, {
+        blockTag: blEvent.blockNumber,
+      });
+
+      if (!newL1OptEscrowBalance.eq(l1OptEscrowBalance) || !newL1ArbEscrowBalance.eq(l1ArbEscrowBalance)) {
+        l1OptEscrowBalance = newL1OptEscrowBalance;
+        l1ArbEscrowBalance = newL1ArbEscrowBalance;
+        findings.push(l1Finding(l1OptEscrowBalance.toString(), l1ArbEscrowBalance.toString()));
+        emitL1Alert(l1OptEscrowBalance.toString(), l1ArbEscrowBalance.toString());
```

Comment:
This should not be used at all in the production code

Reply:
@Vxatz could you please check now if it looks ok? I still have those lines but they shouldn't be mocks, so I'm not sure why they shouldn't be there

Reply:
My comment was actually only about L39, GH shows displays more lines for some reason.

In any case it now LGTM:)